langchain/src/chat_models/baiduwenxin.ts

```diff
@@ -0,0 +1,459 @@
+  messages: WenxinMessage[];
+  stream?: boolean;
+  user_id?: string;
+  temperature?: number;
+  top_p?: number;
+  penalty_score?: number;
+}
+
+interface ChatCompletionResponse {
+  id: string;
+  object: string;
+  created: number;
+  result: string;
+  need_clear_history: boolean;
+  usage: TokenUsage;
+}
+
+declare interface BaiduWenxinChatInput {
+  /** Model name to use
+   * @default "ERNIE-Bot-turbo"
+   */
+  modelName: string;
+
+  /** Whether to stream the results or not. Defaults to false. */
+  streaming?: boolean;
+
+  /** Messages to pass as a prefix to the prompt */
+  prefixMessages?: WenxinMessage[];
+
+  /**
+   * ID of the end-user who made requests.
+   */
+  userId?: string;
+
+  /**
+   * API key to use when making requests. Defaults to the value of
+   * `BAIDU_API_KEY` environment variable.
+   */
+  baiduApiKey?: string;
+
+  /**
+   * Secret key to use when making requests. Defaults to the value of
+   * `BAIDU_SECRET_KEY` environment variable.
+   */
+  baiduSecretKey?: string;
+
+  /** Amount of randomness injected into the response. Ranges
+   * from 0 to 1 (0 is not included). Use temp closer to 0 for analytical /
+   * multiple choice, and temp closer to 1 for creative
+   * and generative tasks. Defaults to 0.95.
+   * Only supported for `modelName` of `WenxinModelName.ERNIE_BOT`.
+   */
+  temperature?: number;
+
+  /** Total probability mass of tokens to consider at each step. Range
+   * from 0 to 1.0. Defaults to 0.8.
+   * Only supported for `modelName` of `WenxinModelName.ERNIE_BOT`.
+   */
+  topP?: number;
+
+  /** Penalizes repeated tokens according to frequency. Range
+   * from 1.0 to 2.0. Defaults to 1.0.
+   * Only supported for `modelName` of `WenxinModelName.ERNIE_BOT`.
+   */
+  penaltyScore?: number;
+}
+
+function messageTypeToWenxinRole(type: MessageType): WenxinMessageRole {
+  switch (type) {
+    case "ai":
+      return "assistant";
+    case "human":
+      return "user";
+    case "system":
+      throw new Error("System messages not supported");
+    default:
+      throw new Error(`Unknown message type: ${type}`);
+  }
+}
+
+/**
+ * Wrapper around Baidu ERNIE large language models that use the Chat endpoint.
+ *
+ * To use you should have the `BAIDU_API_KEY` and `BAIDU_SECRET_KEY`
+ * environment variable set.
+ *
+ * @augments BaseLLM
+ * @augments BaiduERNIEInput
+ */
+export class ChatBaiduWenxin
+  extends BaseChatModel
+  implements BaiduWenxinChatInput
+{
+  declare CallOptions: BaseLanguageModelCallOptions;
+
+  get callKeys(): string[] {
+    return ["stop", "signal", "options"];
+  }
+
+  get lc_secrets(): { [key: string]: string } | undefined {
+    return {
+      baiduApiKey: "BAIDU_API_KEY",
+      baiduSecretKey: "BAIDU_SECRET_KEY",
+    };
+  }
+
+  get lc_aliases(): { [key: string]: string } | undefined {
+    return undefined;
+  }
+
+  lc_serializable = true;
+
+  baiduApiKey?: string;
+
+  baiduSecretKey?: string;
+
+  accessToken: string;
+
+  streaming?: boolean;
```

Comment:
nit: Initialize as `streaming = false;`

Reply:
I have a question about this.

ERNIE has default values for most of its params. Should I define them in code as same as ERNIE's setting, or just leave them as `undefined` so that ERNIE can choose the values? Streaming is just in this case too.

Reply:
Because streaming requires a fundamental shift in how the end user handles output, I'd say it's best to have `streaming` explicitly stated. Other values are more ok to have as default.

Reply:
done with all values